pallets/attesters/src/lib.rs

```diff
@@ -0,0 +1,1232 @@
+    pub type CurrentCommittee<T: Config> = StorageValue<_, Vec<T::AccountId>, ValueQuery>;
+
+    #[pallet::storage]
+    pub type PreviousCommittee<T: Config> = StorageValue<_, Vec<T::AccountId>, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn sorted_nominated_attesters)]
+    pub type SortedNominatedAttesters<T: Config> =
+        StorageValue<_, Vec<(T::AccountId, BalanceOf<T>)>, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn active_set)]
+    pub type ActiveSet<T: Config> = StorageValue<_, Vec<T::AccountId>, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn pending_slashes)]
+    pub type PendingSlashes<T: Config> =
+        StorageMap<_, Identity, T::AccountId, Vec<Slash<T::BlockNumber>>>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn pending_unnominations)]
+    pub type PendingUnnominations<T: Config> = StorageMap<
+        _,
+        Blake2_128Concat,
+        T::AccountId,
+        Vec<(T::AccountId, BalanceOf<T>, BlockNumberFor<T>)>,
+    >;
+
+    #[pallet::storage]
+    #[pallet::getter(fn attestations)]
+    pub type Attestations<T: Config> =
+        StorageMap<_, Identity, T::Hash, Attestation<T::AccountId, Vec<u8>>>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn batches)]
+    pub type Batches<T: Config> =
+        StorageMap<_, Identity, [u8; 4], Vec<Batch<T::AccountId, Vec<u8>, T::BlockNumber>>>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn nominations)]
+    pub type Nominations<T: Config> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        T::AccountId, // Attester
+        Blake2_128Concat,
+        T::AccountId, // Nominator
+        BalanceOf<T>,
+    >;
+
+    #[pallet::storage]
+    #[pallet::getter(fn fast_confirmation_cost)]
+    pub type FastConfirmationCost<T: Config> =
+        StorageMap<_, Blake2_128Concat, [u8; 4], BalanceOf<T>>;
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        AttesterRegistered(T::AccountId),
+        AttestationSubmitted(T::AccountId),
+        Nominated(T::AccountId, T::AccountId, BalanceOf<T>),
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        InvalidSignature,
+        InvalidMessage,
+        AlreadyRegistered,
+        PublicKeyMissing,
+        AttestationSignatureInvalid,
+        AttestationDoubleSignAttempt,
+        NotActiveSet,
+        NotInCurrentCommittee,
+        NotRegistered,
+        NoNominationFound,
+        AlreadyNominated,
+        NominatorNotEnoughBalance,
+        NominatorBondTooSmall,
+        AttesterBondTooSmall,
+        MissingNominations,
+        BatchNotFound,
+        BatchAlreadyCommitted,
+        CommitteeSizeTooLarge,
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::weight(10_000)]
+        pub fn register_attester(
+            origin: OriginFor<T>,
+            self_nominate_amount: BalanceOf<T>,
+            ecdsa_key: [u8; 33],
```

Comment:
What's that extra byte (33) for ECDSA pubkeys?

Reply:
https://wiki.hyperledger.org/display/BESU/SECP256R1+Support 

> For the secp256r1 curve, the private key is 256-bit integer (32 bytes) and the compressed public key is 257-bit integer (~ 33 bytes).